### PR TITLE
Huge re-org an a ton of fixes

### PR DESCRIPTION
**Pull request by Carter12s**

### CODE REVIEW

File: roslibrust/src/rosapi/mod.rs [hunk 1]
```diff
@@ -6,8 +6,10 @@
 use crate::{ClientHandle, RosLibRustResult};
 use async_trait::async_trait;
 
-pub mod types;
-use types::*;
+// TODO Fix this... this sucks
+roslibrust_codegen_macro::find_and_generate_ros_messages_relative_to_manifest_dir!(
+    "../assets/ros1_common_interfaces/rosapi"
```
Comment (ssnover) [inline roslibrust/src/rosapi/mod.rs:11]: Can you not use concat! and env!("CARGO_MANIFEST_DIR")?

Comment (ssnover) [reply roslibrust/src/rosapi/mod.rs:11]: For reference: https://doc.rust-lang.org/cargo/reference/environment-variables.html

Comment (Carter12s) [reply roslibrust/src/rosapi/mod.rs:11]: I cant!

At least it didn't seem to work when I tried

As far as I can tell the proc_macro is literally parsting the tokens passed into it as a vec of paths, and I can't pass in evaluatable code? am I crazzy?

Comment (ssnover) [reply roslibrust/src/rosapi/mod.rs:11]: Oh yeah it looks like I did this with the function backed by this macro, not the macro itself.

**Pull request merged**